File: day 1/main.py
"""
Advent of Code Day 1 Part 1
"""      

# read the input text file of the given data
with open('day1\input.txt', 'r') as f:
    lines = f.readlines()
    # read each line and store in an array named data
    data = [int(entry.strip()) for entry in lines]


# set the first value, and set the count to be 0
first_val = data[0]
count = 0
# iterate in the entire list and check each value
# increase the counter by one if it satisfy the criteria
for entry in data[1:]:
    if entry > first_val:
        count += 1
    first_val = entry



print("Solution of Part 1 is",count)



"""
Advent of Code Part 2 
"""

import numpy as np

sliding_windows = np.convolve(data, np.ones(3), 'valid')

first_val = sliding_windows[0]
count2 = 0
for entry in sliding_windows[1:]:
    if entry > first_val:
        count2 += 1
    first_val = entry

print("The measurement",count2) 

File: day 1/main.js
// Get the input in an array form
// from the text fil saved


const fs = require('fs');

fs.readFile('input.txt','utf-8',(err, data) => {
    const input = data.split('\n').map((x) => Number(x))
    const valuesToCheck= []
    let count = 0
    for (let i = 0; i < input.length; i++) {
        valuesToCheck.push(input.slice(i, i+3).reduce((acc,v)=> acc + v, 0))

    }
    for (let i =0; i < valuesToCheck.length; i++) {
        const first = valuesToCheck[i]
        const second = valuesToCheck[i+1]
        if (second > first) {
            ++count
        }
    }

    console.log('Count=', count)
})